TimerStartEventTest.shouldNotReTriggerTimerAfterDuplicateDeployment is flaky

**Summary**

- How often does the test fail? at least once
- Does it block your work? no
- Do we suspect that it is a real failure? considering the major refactoring effort we underwent, probably

**Failures**

Failed build: https://ci.zeebe.camunda.cloud/blue/organizations/jenkins/camunda-cloud%2Fzeebe/detail/np-fix-snapshot-abort/1/tests

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 
[Expect to trigger timer start events only for new deployments, but duplicate deployment causes retriggering of timer start event.] 
Expected size: 2 but was: 1 in:
[{"valueType":"PROCESS_INSTANCE","key":2251799813685256,"position":23,"timestamp":1620223964162,"recordType":"COMMAND","intent":"ACTIVATE_ELEMENT","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","brokerVersion":"1.0.0","value":{"version":2,"processDefinitionKey":2251799813685253,"elementId":"process","bpmnElementType":"PROCESS","bpmnProcessId":"process","processInstanceKey":2251799813685256,"flowScopeKey":-1,"parentProcessInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":17}]
	at io.camunda.zeebe.engine.processing.timer.TimerStartEventTest.shouldNotReTriggerTimerAfterDuplicateDeployment(TimerStartEventTest.java:279)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
</pre>
</details>

**Logs**

<details><summary>Logs</summary>
<pre>
14:09:43.066 [Broker-0-LogStream-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
14:09:43.359 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Recovering state of partition 1 from snapshot
14:09:43.360 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@451b61f2)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@4bdc8721, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@4fe30a82, configuration: Configuration(false)]
14:09:43.363 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
14:09:43.364 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@ae18d1a)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@14eb6945, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@3ba10176, configuration: Configuration(false)]
14:09:43.366 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@6c41bb9a)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@6ace7255, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@25208f08, configuration: Configuration(false)]
14:09:43.488 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@6241a7fd)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@204ddeb7, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@8b8f844, configuration: Configuration(false)]
14:09:43.864 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@544eb569)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@75adb05f, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@36129d5a, configuration: Configuration(false)]
14:09:43.995 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@4d32e4b)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@3afb61fb, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@3fff4efc, configuration: Configuration(false)]
14:09:45.883 [] INFO  io.camunda.zeebe.test - Test failed, following records were exported:
14:09:46.201 [] INFO  io.camunda.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  P[partitionId]K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C DEPLOYMENT CREATE         - #01-> -1  -1 - 
E PROC       CREATED        - #02->#01 K01 - process.xml -> "process" (version:1)
E TIMER      CREATED        - #03->#01 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669
E DEPLOYMENT CREATED        - #04->#01 K02 - process.xml
E DEPLOYMENT FULLY_DISTR    - #05->#01 K02 - 
C DEPLOYMENT CREATE         - #06-> -1  -1 - 
E DEPLOYMENT CREATED        - #07->#06 K04 - process.xml
E DEPLOYMENT FULLY_DISTR    - #08->#06 K04 - 
C DEPLOYMENT CREATE         - #09-> -1  -1 - 
E PROC       CREATED        - #10->#09 K05 - process.xml -> "process" (version:2)
C TIMER      CANCEL         - #11->#09 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669
E TIMER      CREATED        - #12->#09 K07 -  @"start_6"[ -1]  in <process "K05"[?]> due T14:11:59.087
E DEPLOYMENT CREATED        - #13->#09 K06 - process.xml
E DEPLOYMENT FULLY_DISTR    - #14->#09 K06 - 
C TIMER      TRIGGER        - #15->#09 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669
C TIMER      TRIGGER        - #16->#09 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669
C TIMER      TRIGGER        - #17->#09 K07 -  @"start_6"[ -1]  in <process "K05"[?]> due T14:11:59.087
E TIMER      CANCELED       - #18->#11 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669
R TIMER      TRIGGER        - #19->#15 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669 !NOT_FOUND (Expected to trigger timer with key '2251799813685251', but no such timer was found)
R TIMER      TRIGGER        - #20->#16 K03 -  @"start_5"[ -1]  in <process "K01"[?]> due T14:09:58.669 !NOT_FOUND (Expected to trigger timer with key '2251799813685251', but no such timer was found)
E TIMER      TRIGGERED      - #21->#17 K07 -  @"start_6"[ -1]  in <process "K05"[?]> due T14:11:59.087
E PROC_EVNT  TRIGGERING     - #22->#17 K09 -  @"start_6"[K05] in <process K05[K08]> (no vars)
C PROC_INST  ACTIVATE       - #23->#17 K08 - PROCESS "process" in <process "process"[K08]>
E PROC_INST  ACTIVATING     - #24->#23 K08 - PROCESS "process" in <process "process"[K08]>
E PROC_INST  ACTIVATED      - #25->#23 K08 - PROCESS "process" in <process "process"[K08]>
E PROC_EVNT  TRIGGERED      - #26->#23 K09 -  @"start_6"[K05] in <process K05[K08]> (no vars)
C PROC_INST  ACTIVATE       - #27->#23 K10 - START_EVENT "start_6" in <process "process"[K08]>
E PROC_INST  ACTIVATING     - #28->#27 K10 - START_EVENT "start_6" in <process "process"[K08]>
E PROC_INST  ACTIVATED      - #29->#27 K10 - START_EVENT "start_6" in <process "process"[K08]>
C PROC_INST  COMPLETE       - #30->#27 K10 - START_EVENT "start_6" in <process "process"[K08]>
E PROC_INST  COMPLETING     - #31->#30 K10 - START_EVENT "start_6" in <process "process"[K08]>
E PROC_INST  COMPLETED      - #32->#30 K10 - START_EVENT "start_6" in <process "process"[K08]>
E PROC_INST  SEQ_FLOW_TAKEN - #33->#30 K11 - SEQUENCE_FLOW "sequenc..a0c8407" in <process "process"[K08]>
C PROC_INST  ACTIVATE       - #34->#30 K12 - END_EVENT "end_5" in <process "process"[K08]>
E PROC_INST  ACTIVATING     - #35->#34 K12 - END_EVENT "end_5" in <process "process"[K08]>
E PROC_INST  ACTIVATED      - #36->#34 K12 - END_EVENT "end_5" in <process "process"[K08]>
E PROC_INST  COMPLETING     - #37->#34 K12 - END_EVENT "end_5" in <process "process"[K08]>
E PROC_INST  COMPLETED      - #38->#34 K12 - END_EVENT "end_5" in <process "process"[K08]>
C PROC_INST  COMPLETE       - #39->#34 K08 - PROCESS "process" in <process "process"[K08]>
E PROC_INST  COMPLETING     - #40->#39 K08 - PROCESS "process" in <process "process"[K08]>
E PROC_INST  COMPLETED      - #41->#39 K08 - PROCESS "process" in <process "process"[K08]>
-------------- Deployed Processes ----------------------
process.xml -> "process" (version:1)[K01] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_6056b6ed-d535-4a7d-8c46-b3a459c69133" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process" isExecutable="true">
    <startEvent id="start_5" name="start_5">
      <outgoing>sequenceFlow_3beab205-c09e-4d86-b8ab-1c6d0ef03c61</outgoing>
      <timerEventDefinition id="timerEventDefinition_67f17bd1-8442-4869-9704-13d0c8cf95d3">
        <timeDate id="timeDate_ab7ad143-98a1-4559-b8e2-7972ac6c95d9">=now() + duration("PT15S")</timeDate>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="end_5" name="end_5">
      <incoming>sequenceFlow_3beab205-c09e-4d86-b8ab-1c6d0ef03c61</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_3beab205-c09e-4d86-b8ab-1c6d0ef03c61" sourceRef="start_5" targetRef="end_5"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_45e6e417-983c-4620-8bd7-cf1729ac5a7d">
    <bpmndi:BPMNPlane bpmnElement="process" id="BPMNPlane_4ed7e933-aa69-48a5-bf60-a631c32780ec">
      <bpmndi:BPMNShape bpmnElement="start_5" id="BPMNShape_d4add8ca-d5ef-42fa-a4e0-d097c5f28497">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="end_5" id="BPMNShape_2728b6fc-7e4b-40ad-9218-ef31bf1dd1f2">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_3beab205-c09e-4d86-b8ab-1c6d0ef03c61" id="BPMNEdge_4bf91e2b-8096-4f9c-911e-0dc77eee75ac">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>
process.xml -> "process" (version:2)[K05] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_a5ddb34e-3bc6-407f-9fcb-e2ba9ec1ab5c" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process" isExecutable="true">
    <startEvent id="start_6" name="start_6">
      <outgoing>sequenceFlow_8081f257-755a-443b-8d9a-8d38ca0c8407</outgoing>
      <timerEventDefinition id="timerEventDefinition_c73e7430-d4af-41df-a9ef-3670e182734d">
        <timeDate id="timeDate_806cc537-fae8-4e8b-ae5d-cb9af4342700">=now() + duration("PT15S")</timeDate>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="end_5" name="end_5">
      <incoming>sequenceFlow_8081f257-755a-443b-8d9a-8d38ca0c8407</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_8081f257-755a-443b-8d9a-8d38ca0c8407" sourceRef="start_6" targetRef="end_5"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_094323bd-e8d2-47f8-8719-6b4afb9f30bc">
    <bpmndi:BPMNPlane bpmnElement="process" id="BPMNPlane_d0de8edd-00f7-450d-bdbd-cc35b6d64fa5">
      <bpmndi:BPMNShape bpmnElement="start_6" id="BPMNShape_2839cd2a-9bbf-4ea0-bbfb-3ed4f962e98e">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="end_5" id="BPMNShape_802339ee-2ab4-48e3-9d4c-6727e3b70dfe">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_8081f257-755a-443b-8d9a-8d38ca0c8407" id="BPMNEdge_1f08e776-f398-410d-b66d-5039ed7738ff">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>
--------------- Decomposed keys (for debugging) -----------------
 -1 <-> -1
K01 <-> 2251799813685249
K02 <-> 2251799813685250
K03 <-> 2251799813685251
K04 <-> 2251799813685252
K05 <-> 2251799813685253
K06 <-> 2251799813685254
K07 <-> 2251799813685255
K08 <-> 2251799813685256
K09 <-> 2251799813685257
K10 <-> 2251799813685258
K11 <-> 2251799813685259
K12 <-> 2251799813685260
14:09:46.262 [] DEBUG io.camunda.zeebe.util.buffer - Close stream processor
14:09:46.263 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
14:09:46.359 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream-1
14:09:46.360 [stream-1-write-buffer] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
14:09:46.360 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream-1 close 2 readers
14:09:46.361 [] DEBUG io.camunda.zeebe.broker.test - Clean up test files on path /tmp/junit13101847059063081682
14:09:46.361 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
14:09:46.362 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
14:09:46.363 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
14:09:46.363 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
</pre>
</details>

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java
@@ -234,12 +234,16 @@ public final class TimerStartEventTest {
             .getProcessesMetadata()
             .get(0);
 
-    RecordingExporter.timerRecords(TimerIntent.CREATED)
-        .withProcessDefinitionKey(deployedProcess.getProcessDefinitionKey())
-        .await();
+    engine.awaitProcessingOf(
+        RecordingExporter.timerRecords(TimerIntent.CREATED)
+            .withProcessDefinitionKey(deployedProcess.getProcessDefinitionKey())
+            .getFirst());
 
     // when
     engine.increaseTime(Duration.ofMinutes(1));
+    RecordingExporter.timerRecords(TimerIntent.TRIGGER)
+        .withProcessDefinitionKey(deployedProcess.getProcessDefinitionKey())
+        .await();
     engine.deployment().withXmlResource(firstVersion).deploy();
     engine.increaseTime(Duration.ofMinutes(1));
 
@@ -252,10 +256,14 @@ public final class TimerStartEventTest {
             .getValue()
             .getProcessesMetadata()
             .get(0);
-    RecordingExporter.timerRecords(TimerIntent.CREATED)
+    engine.awaitProcessingOf(
+        RecordingExporter.timerRecords(TimerIntent.CREATED)
+            .withProcessDefinitionKey(secondVersionMetadata.getProcessDefinitionKey())
+            .getFirst());
+    engine.increaseTime(Duration.ofMinutes(1));
+    RecordingExporter.timerRecords(TimerIntent.TRIGGER)
         .withProcessDefinitionKey(secondVersionMetadata.getProcessDefinitionKey())
         .await();
-    engine.increaseTime(Duration.ofMinutes(1));
 
     // expect only two process instances created by a timer
     final var processInstanceRecords =

--- a/engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java
@@ -73,6 +73,7 @@ import org.agrona.DirectBuffer;
 import org.agrona.MutableDirectBuffer;
 import org.agrona.collections.Int2ObjectHashMap;
 import org.agrona.concurrent.UnsafeBuffer;
+import org.awaitility.Awaitility;
 import org.junit.rules.ExternalResource;
 import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
@@ -368,6 +369,13 @@ public final class EngineRule extends ExternalResource {
                                   MsgPackConverter.convertToJson(value.getDirectBuffer())));
                   return entries;
                 }));
+  }
+
+  public void awaitProcessingOf(final Record<?> record) {
+    final var recordPosition = record.getPosition();
+
+    Awaitility.await("await the record to be processed")
+        .until(() -> getLastProcessedPosition() >= recordPosition);
   }
 
   private static final class VersatileBlob implements DbKey, DbValue {

--- a/engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java
@@ -8,6 +8,7 @@
 package io.camunda.zeebe.engine.util;
 
 import static io.camunda.zeebe.test.util.record.RecordingExporter.jobRecords;
+import static org.assertj.core.api.Assertions.assertThat;
 
 import io.camunda.zeebe.db.DbKey;
 import io.camunda.zeebe.db.DbValue;
@@ -374,8 +375,16 @@ public final class EngineRule extends ExternalResource {
   public void awaitProcessingOf(final Record<?> record) {
     final var recordPosition = record.getPosition();
 
-    Awaitility.await("await the record to be processed")
-        .until(() -> getLastProcessedPosition() >= recordPosition);
+    Awaitility.await(
+            String.format(
+                "Await the %s.%s to be processed at position %d",
+                record.getValueType(), record.getIntent(), recordPosition))
+        .untilAsserted(
+            () ->
+                assertThat(getLastProcessedPosition())
+                    .describedAs(
+                        "Last process position should be greater or equal to " + recordPosition)
+                    .isGreaterThanOrEqualTo(recordPosition));
   }
 
   private static final class VersatileBlob implements DbKey, DbValue {

